app/models/miq_action.rb

```diff
@@ -848,6 +848,31 @@ def action_container_image_analyze(action, rec, inputs)
     rec.scan
   end
 
+  def action_container_image_scan_all(action, rec, _inputs)
+    unless rec.kind_of?(ContainerImage)
+      MiqPolicy.logger.error("MIQ(#{__method__}): Unable to perform action [#{action.description}],"\
+                             " object [#{rec.inspect}] is not a Container Image")
+      return
+    end
+    ext = rec.ext_management_system
+
+    unless ext
+      MiqPolicy.logger.error("MIQ(#{__method__}): Unable to perform action [#{action.description}],"\
+                             " object [#{rec.inspect}] does not belong to an ems")
+      return
+    end
+    ext.container_images.each do |container_image|
+      MiqQueue.put(
+        :class_name  => container_image.class.name,
+        :instance_id => container_image.id,
+        :method_name => 'scan',
+        :args        => [],
+        :deliver_on  => Time.now.utc + 3.minutes,
```

Comment:
Would it be better to use a task_id value in the MiqQueue to ensure the scans are done synchronously?  How did you come up with 3 minutes?